core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -53,81 +57,115 @@
-            new HubCategoryConfiguration(),
-            new LoggingCategoryConfiguration(),
-            new PredictionsCategoryConfiguration(),
-            new StorageCategoryConfiguration()
-        );
-
-        final Map<String, CategoryConfiguration> actualConfigs = new HashMap<>();
-        try {
-            for (CategoryConfiguration possibleConfig : possibleConfigs) {
-                String categoryJsonKey = possibleConfig.getCategoryType().getConfigurationKey();
-                if (json.has(categoryJsonKey)) {
-                    possibleConfig.populateFromJSON(json.getJSONObject(categoryJsonKey));
-                    actualConfigs.put(categoryJsonKey, possibleConfig);
-                }
-            }
-        } catch (JSONException error) {
-            throw new AmplifyException(
-                "Could not parse amplifyconfiguration.json ",
-                error, "Check any modifications made to the file."
-            );
-        }
-        return new AmplifyConfiguration(Immutable.of(actualConfigs));
+        return builder(json).build();
     }
 
     /**
      * Creates an AmplifyConfiguration from an amplifyconfiguration.json file.
+     *
      * @param context Context needed for reading JSON file
      * @return An Amplify configuration instance
      * @throws AmplifyException If there is a problem in the config file
+     * @deprecated This method is now deprecated in favor of
+     *          {@link AmplifyConfiguration#builder(Context)}.
      */
     @SuppressWarnings("WeakerAccess")
+    @Deprecated
     @NonNull
     public static AmplifyConfiguration fromConfigFile(@NonNull Context context) throws AmplifyException {
-        return fromConfigFile(context, getConfigResourceId(context));
+        return builder(context, getConfigResourceId(context)).build();
     }
 
     /**
      * Creates an AmplifyConfiguration from a particular configuration file.
+     *
      * @param context Android Context
      * @param configFileResourceId
      *        The Android resource ID of a raw resource which contains
      *        an amplify configuration as JSON
      * @return An Amplify configuration instance
      * @throws AmplifyException If there is a problem in the config file
+     * @deprecated This method is now deprecated in favor of
+     *          {@link AmplifyConfiguration#builder(Context, int)}.
      */
+    @Deprecated
     @NonNull
     public static AmplifyConfiguration fromConfigFile(
             @NonNull Context context, @RawRes int configFileResourceId) throws AmplifyException {
-        return fromJson(readInputJson(context, configFileResourceId));
+        return builder(context, configFileResourceId).build();
+    }
+
+    /**
+     * Obtain a map of additional platforms that are using this library
+     * for tracking usage metrics.
+     * @return A map of additional platforms and their versions.
+     */
+    @NonNull
+    public Map<Platform, String> getPlatformVersions() {
```

Comment:
Order matters right? `LinkedHashMap` as a return type here? Also: will `Immutable` respect that? Or does it return a generic `Map`? May need to use generics and change it to liek:
```java
public static <T, M extends Map<T>> M of(M map) {
    ...
}
```